src/sisl_toolbox/siesta/minimizer/_minimize.py

```diff
@@ -288,15 +296,18 @@ def _minimize_func(self, norm_variables, *args):
             pass
 
         # Else we have to call minimize
-        metric = np.array(self(variables, *args))
-        # add the data to the output file and hash it
-        self._fh.write_data(
-            variables.reshape(-1, 1), metric.reshape(-1, 1), fmt="20.17e"
-        )
-        self._fh.flush()
-        self.data.x.append(variables)
-        self.data.y.append(metric)
-        self.data.hash.append(current_hash)
+        metric = self(variables, *args)
+
+        if metric is not None:
```

Comment:
when would `metric` be none? Would it even be viable?

Shouldn't something break if the metric is none?

Reply:
I think None meant that it has not been able to compute the metric, e.g. the calculation has failed. It shouldn't break because the optimizer can set variable values that result in a failed calculation.

Reply:
I recall having tested with all the minimizers and they all handled a None, but I can't say 100%